apps/charterafrica/src/lib/ecosystem/airtable/processData.js

```diff
@@ -47,27 +47,32 @@ function mapSupporterIdsToObjects(supporterIds, config, { partnersData }) {
   return mapped.filter(Boolean);
 }
 
-function mapSocialMediaIdsToObjects(socialMedia, config, tableData) {
-  const {
-    schema: { socialMediaTableColumns },
-  } = config;
-  const { name, url } = socialMediaTableColumns;
-  const { socialMediaData = [] } = tableData;
-  const mapped = socialMedia.map((id) => {
-    const { fields } = socialMediaData.find((item) => item.id === id) || {};
-
-    if (!fields) {
+function mapSocialMediaColumnsToObjects(columns, data) {
+  const getData = (name) => {
+    const link = getValue(data, columns[name]);
+    if (!link) {
       return null;
     }
     return {
-      name: getValue(fields, name),
-      link: getValue(fields, url),
+      name,
+      link,
     };
-  });
-  return mapped.filter(Boolean);
+  };
+  return [
+    getData("twitter"),
+    getData("facebook"),
+    getData("youtube"),
+    getData("slack"),
+    getData("tiktok"),
+    getData("linkedIn"),
+    getData("instagram"),
+    getData("telegram"),
+    getData("discord"),
+    getData("whatsapp"),
+  ].filter(Boolean);
```

Comment:
We can declare supported social media platforms somewhere at the top of the file
```js
const SUPPORTED_SOCIAL_MEDIA = "twitter,facebook,youtube,slack,tiktok,linkedIn,instagram,telegram,discord,whatsapp";
```

We can also use array if you prefer that.

```suggestion
  return SUPPORTED_SOCIAL_MEDIA.split(",").map(getData).filter(Boolean);
```